Add bonded devices support

pdecat opened the pull request:
This PR adds support for bonded devices such as the [RAPT Bluetooth Thermometer](https://kegland.com.au/products/rapt-bluetooth-thermometer-20-to-300c-20cm-htc-probe):

<img width="1320" height="910" alt="image" src="https://github.com/user-attachments/assets/e1f83583-f246-43ac-a0f8-ed759fdf753b" />

<img width="1442" height="1027" alt="image" src="https://github.com/user-attachments/assets/5685d98a-9df2-407b-9ddc-d0838e80f371" />

pdecat committed:
Add bonded devices support

pdecat commented:
FWIW, I started [my own RAPT API Home Assistant integration](https://github.com/pdecat/rapt-io-ha-integration) a few months ago as none existed at that time, and only resumed working on it this morning to make it functional. It's only when I went to install it in my main Home Assistant instance via HACS that I discovered your integration :melting_face:

pdecat committed:
Add firmware version to device information

pdecat committed:
Add device type to model for bonded devices

pdecat commented on custom_components/rapt_cloud_link/sensor.py:
```diff
@@ -48,14 +60,10 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry, async_add_e
-            name_suffix="Temperature",
-            unique_suffix="temperature",
-            unit="°C"
+            coordinator, device_id, model="BrewZilla", name_suffix="Temperature", unique_suffix="temperature", unit="°C"
```
Sorry about these formatting changes, can't revert them if you want.

berra200 commented:
Hi! Thanks a lot for your contribution 🙏 Really cool to hear that you had started your own RAPT API integration earlier — I totally get the surprise of discovering someone else’s project at the same time 😅

I noticed that your branch is about 5 commits behind main. Could you update it with the latest changes from main (via git merge or git rebase) and check that everything still works? That would make it much easier for me to review and test your changes.

Looking forward to seeing your additions — it’s awesome to have more Kegland devices supported! 🍻

pdecat commented:
Hi @berra200, thanks! Rebased on develop and updated the PR to target that branch.

Still working great:

<img width="1920" height="1014" alt="image" src="https://github.com/user-attachments/assets/edcbd9b3-6bd7-42f9-81a4-3adec48bd245" />

<img width="1924" height="1125" alt="image" src="https://github.com/user-attachments/assets/07a562ef-20e3-417d-9f5f-6323f4599b65" />

berra200 commented:
Thanks a lot for rebasing and updating the PR 🙏 I’ve merged it into develop now and will test it later today before pushing to main. Really appreciate your contribution!

pdecat commented:
Thanks @berra200!

Just noticed I forgot to update the features section of the README https://github.com/berra200/home-assistant-rapt-cloud-link?tab=readme-ov-file#features

berra200 commented:
Good point, thanks for noticing! 🙌 I’ve now merged everything into main, drafted a new release, and updated both the README and version numbers.
